File: src/main/kotlin/cz/vernjan/ctf/he21/CafeShop.java
package cz.vernjan.ctf.he21;

import com.google.common.hash.Hashing;

import java.math.BigInteger;
import java.nio.charset.StandardCharsets;

public class CafeShop {

    public static void main(String[] args) {

        String productName = "Vanilla Cafe";
        String id = "11865457";

        String sha256hex = Hashing.sha256()
                .hashString(productName, StandardCharsets.UTF_8)
                .toString();

        System.out.println(sha256hex);

        System.out.println(new BigInteger(sha256hex, 16));

        System.out.println(Hashing.sha256()
                .hashString(productName, StandardCharsets.UTF_8).asInt());

    }

}

File: src/main/kotlin/cz/vernjan/ctf/he21/CafeShop.kt
@file:Suppress("UnstableApiUsage")

package cz.vernjan.ctf.he21

import com.google.common.hash.Hashing
import cz.vernjan.ctf.toHex

fun main() {
    val adjectives = listOf("Yummy", "Sweet", "Cherry", "Groovy", "Chocolate")
    val candies = listOf("Cake", "Doughnut", "Pie", "Lollipop", "Gum")

    for (adjective in adjectives) {
        for (candy in candies) {
            println("Testing $adjective $candy ..")
            for (i in 0..10_000_000) {
                val number = i.toString().padStart(8, '0')
                val product = "$number $adjective $candy"
                val hash = Hashing.sha256().hashBytes(product.toByteArray()).asBytes().toHex()
                if (hash.contains("c01a") && hash.contains("decaf")) {
                    println(">>> $product ($hash)")
                }
            }
        }
    }
}